network/src/peer_store/sqlite/peer_store.rs

```diff
@@ -146,44 +144,59 @@ impl SqlitePeerStore {
         }
 
         let tx = self.conn.transaction().expect("db tx");
-        db::PeerInfo::delete(&tx, candidate_peer.id).expect("delete peer error");
-        db::PeerAddr::delete_by_peer_id(&tx, candidate_peer.id)
+        db::PeerInfoDB::delete(&tx, &candidate_peer.peer_id).expect("delete peer error");
+        db::PeerAddrDB::delete_by_peer_id(&tx, &candidate_peer.peer_id)
             .expect("delete peer by peer_id error");
         tx.commit().expect("delete peer error");
         Ok(())
     }
 
-    fn fetch_peer_info(&self, peer_id: &PeerId) -> db::PeerInfo {
+    /// check and try delete peer_addr if peer_addr reach limit
+    fn check_peer_addrs_limit(&mut self, peer_id: &PeerId) -> Result<(), ()> {
+        let now = unix_time_as_millis();
+        let peer_addrs_count = db::PeerAddrDB::count(&self.conn, peer_id).expect("peer info count");
+        if peer_addrs_count < MAX_ADDRS {
+            return Ok(());
+        }
+        let mut peer_addrs =
+            db::PeerAddrDB::get_addrs(&self.conn, peer_id, MAX_ADDRS).expect("peer addrs");
+        let mut terrible_addrs_count = 0;
+        for paddr in &peer_addrs {
+            if paddr.is_terrible(now) {
+                db::PeerAddrDB::delete(&self.conn, &paddr.peer_id, &paddr.addr)
+                    .expect("delete peer addr");
+                terrible_addrs_count += 1;
+            }
+        }
+        // have evict addrs
+        if terrible_addrs_count > 0 {
+            return Ok(());
+        }
+        // find oldest last_connected_at_ms addr
+        peer_addrs.sort_by(|a, b| a.last_connected_at_ms.cmp(&b.last_connected_at_ms));
+        if let Some(old_peer_addr) = peer_addrs.get(0) {
+            db::PeerAddrDB::delete(&self.conn, &old_peer_addr.peer_id, &old_peer_addr.addr)
+                .expect("evict peer addr");
+        }
+        Ok(())
```

Comment:
Where is the error case

Reply:
There no error case, the intention to use Result is because `bool` return is confused which hard to distinguish whether the check is pass or there is error occurred. use Result().is_err or is_ok is more clear to determine the intention of return value.

another benefit is rustc display warning if we missing to handle the result.

Reply:
It's a mistake, should return ().